### DISCUSSION TITLE
Writing configs

### DISCUSSION THREAD

**Discussion by javierin** [basking-in-the-sun2000/solar-logger#20] [Q&A]
Does anybody know a way to push configuration changes to the inverter through the modbus connection? I am thinking of the following scenario.

I have a LUNA2000 battery. I usually charge it overnight using cheap tarrif in order to cover the first 2-3  hours of consumption at peak rates when there is not enough sun irradiation. However, there are some cloudy days where I wish I would have charge it overnight up to 100% to cover more of that cloudy day consumption. 

The forecast APIs are already there, but I have no clue of any way to configure stuff on the inverter, is it even possible? There must be some way because the FusionSolar app connects there too using the installer login / pass.

**Comment by basking-in-the-sun2000** [comment]
Had hoped someone would have shared some experience about this. I haven't found a way to send commands to the inverter. There are several write registers on the map, but not sure you can change them without authenticating first.

**Reply by andysu** [reply]
i use 0x06 and 0x10 funcions (but in modbusRTU) without any auth.
I haven't been able to use the 0x41 functions yet.
But I hope after the war I will succeed.

**Reply by basking-in-the-sun2000** [reply]
Stay safe and courage. Hope this madness ends soon, and you and your family are safe

In theory it should be possible to do it through tcp, the app does. The file feature is resulting elusive lol

**Comment by andysu** [comment]
**javierin**
please do the following:
1. through fusionsolar make the changes you need
2. make program logs
3. send them to me (adm@elsistema.com.ua) with the exact time when the changes were made (preferably done several times).

**Comment by javierin** [comment]
Andy, by program logs you mean the logs dumped using the FusionSolar App with installer account logged in locally?

Thanks

**Reply by andysu** [reply]
yes
all logs by fusionsolar mobile app. (file zip with inverter sn_datetime)
I don't have physical access to my inverters yet
